### PR TITLE
Add translations-management scripts

### PR DESCRIPTION
**Pull request by a2br**
## Summary
These scripts have been requested in issue https://github.com/NMF-earth/nmf-app/issues/225.
There are 2 of them:
- `group-translation-files.ts`: gets the translations from all the known translation files in the repo to put them all together, sorted by language
- `spread-translation-files.ts`: reverses the previous script's work. It spreads the data from the reference files (built by `group-translation-files.ts`) across all the other translation files in known translation directories

The reference translation files (ex: `en.json`) will be placed in `/scripts/poeditor`, next to the scripts.

They have been built to be flexible in case of future changes. For example, adding a new language only requires to add a 2-letters string in an array -- same for the translation dirs.

## Notes
They have not been tested. Actually, I don't know how to do it. `tsc` doesn't compile anymore, even after some twists in `tsconfig.json`... 🤔 As always, let me know if you spot an error.

## How to run them?
Once compiled using `tsc`, I assume you'd just have to run `node ./path/to/script`. As written above, I haven't tested them though, since the TypeScript Compiler doesn't compile. Tell me if I'm talking nonsense, I'm not familiar with React Native projects nor the way they are executed.

### CODE REVIEW

File: app/utils/translations/poeditor/group.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+export {};
```
Comment (PierreBresson) [inline app/utils/translations/poeditor/group.ts:1]: probably not needed. i get `SyntaxError: Unexpected token 'export'`

File: app/utils/translations/poeditor/group.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import * as fs from "graceful-fs";
```
Comment (PierreBresson) [inline app/utils/translations/poeditor/group.ts:2]: `const fs = require("fs");` like in `generate-translation-files.js`, no need to add a devDep I believe.
If so, use `yarn add -D graceful-fs` and remove the ^ in `package.json`

File: app/utils/translations/poeditor/spread.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
+export {};
```
Comment (PierreBresson) [inline app/utils/translations/poeditor/spread.ts:1]: same applies to this file

**Review by PierreBresson — CHANGES_REQUESTED** [review]
Hi,

Could you place the script files inside `scripts` folder instead? My mistake, it doesn't make sense, therefore I've edited the issue.

Alse, rename the scripts `group-translation-files.js` and `spread-translation-files.js` so it's more obvious.

So far, I couldn't make the script to work

**Comment by PierreBresson** [comment]
Besides that, you don't have to care about `yarn tsc`, these scripts are not meant to be use by the app. They are used when I'm import / exporting translations via Poeditor, not when the app runs.

File: app/utils/translations/poeditor/group.ts (hunk 1 above)
Comment (a2br) [reply app/utils/translations/poeditor/group.ts:1]: Oh, I wrote this out of habit, to make TS consider the file as a module and therefore avoid some unexpected behavior from the compiler; not needed here.

File: app/utils/translations/poeditor/group.ts (hunk 2 above)
Comment (a2br) [reply app/utils/translations/poeditor/group.ts:2]: ESLint doesn't like `require`, I'll see how it does with the CI test.

Comment (a2br) [reply app/utils/translations/poeditor/group.ts:2]: Actually, I'll use `import * as fs from "fs";`, for some reason IntelliSense didn't include `fs` when I wrote the code.

**Comment by a2br** [comment]
I was worrying about that because these are TS files, they need to be compiled to run; since they import stuff from other TS files, I thought all the files would need to be compiled together.

**Review by PierreBresson — CHANGES_REQUESTED** [review]
still can't run the scripts:
`node scripts/poeditor/spread-translation-files.ts` gives the following error : 
```
import * as fs from "fs";
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

**Comment by a2br** [comment]
It means that you're trying to run the script as an ES2015 module, but Node.js supports CommonJS. This is a configuration error of TypeScript or whatever compiles its. To use the js `import` statements instead of the `require` statements, you'll need to add `"type": "module"` in `package.json`, although I recommend you to manually edit the files. As a reminder, you need to compile TypeScript files before you run them. The error you got may also appear when trying to run a TS file (which isn't supposed to be used).

**Comment by a2br** [comment]
`spread-translation-files.ts` didn't need to be in TypeScript, so I changed it to JavaScript. The problem you encountered should be fixed now.

**Comment by a2br** [comment]
I'll keep working on `group-translation-files.ts` to make the same changes tomorrow evening, I'll have to manually import all the JSON files manually

**Comment by PierreBresson** [comment]
@a2br still not working at all. I think you don't really understand the task unfortunately. In the first hand, I don't see the generated files that should be comitted with all the translations. Please re-submit if you find a solution.

**Comment by a2br** [comment]
I have fully understood what you meant. I've been working on them this morning and their are now functional. I'll resolve some path problems and I'll open another PR this evening (with the language JSONs included then)

**Pull request closed without merging**